plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryProxyConfig.java

```diff
@@ -0,0 +1,107 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.ConfigurationException;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestBigQueryProxyConfig
+{
+    @Test
+    public void testDefaults()
+    {
+        assertRecordedDefaults(recordDefaults(BigQueryProxyConfig.class)
+                .setUri(null)
+                .setPassword(null)
+                .setUsername(null)
+                .setKeystorePath(null)
+                .setKeystorePassword(null)
+                .setTruststorePath(null)
+                .setTruststorePassword(null));
+    }
+
+    @Test
+    public void testExplicitPropertyMappings()
+            throws IOException
+    {
+        Path keystoreFile = Files.createTempFile(null, null);
+        Path truststoreFile = Files.createTempFile(null, null);
+
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("bigquery.rpc.http-proxy", "http://localhost:8000")
+                .put("bigquery.rpc.http-proxy.username", "username")
+                .put("bigquery.rpc.http-proxy.password", "password")
+                .put("bigquery.rpc.http-proxy.trust-store-path", truststoreFile.toString())
+                .put("bigquery.rpc.http-proxy.trust-store-password", "password-truststore")
+                .put("bigquery.rpc.http-proxy.key-store-path", keystoreFile.toString())
+                .put("bigquery.rpc.http-proxy.key-store-password", "password-keystore")
+                .buildOrThrow();
+
+        BigQueryProxyConfig expected = new BigQueryProxyConfig()
+                .setUri(URI.create("http://localhost:8000"))
+                .setUsername("username")
+                .setPassword("password")
+                .setKeystorePath(keystoreFile.toFile())
+                .setKeystorePassword("password-keystore")
+                .setTruststorePath(truststoreFile.toFile())
+                .setTruststorePassword("password-truststore");
+
+        assertFullMapping(properties, expected);
+    }
+
+    @Test
+    public void testInvalidConfiguration()
```

Comment:
You could use `TestBigqueryPlugin` for such tests. Such way tests fully guice and stuff. The current test is very low level and may not provide enough coverage as it depends on guice.